haskell/src/Lorentz/Contracts/Spec/FA2Interface.hs

```diff
@@ -240,23 +241,58 @@ type FA2ParameterC param =
 -- | Owner hook interface
 --
 
-type TransferDestinationDescriptor =
-  ("to_" :! Maybe Address
-  , ("token_id" :! TokenId, "amount" :! Natural))
+data TransferDestinationDescriptor = TransferDestinationDescriptor
+  { tddTo :: Maybe Address
+  , tddTokenId :: TokenId
+  , tddAmount :: Natural
+  }
+  deriving stock (Show, Eq)
+
+deriving anyclass instance IsoValue TransferDestinationDescriptor
+deriving anyclass instance HasAnnotation TransferDestinationDescriptor
+$(customGeneric "TransferDestinationDescriptor" $ withDepths
+    [ cstr @0
+      [ fld @1
+      , fld @2
+      , fld @2
+      ]
+    ]
+  )
+
+data TransferDescriptor = TransferDescriptor
+  { tdFrom :: Maybe Address
+  , tdTxs :: [TransferDestinationDescriptor]
+  }
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (IsoValue, HasAnnotation)
 
-type TransferDescriptor =
-  ( "from_" :! Maybe Address
-  , ("txs" :! [TransferDestinationDescriptor]))
+data TransferDescriptorParam = TransferDescriptorParam
+  { tdpFa2 :: Address
+  , tdpBatch :: [TransferDescriptor]
+  , tdpOperator :: Address
+  }
+  deriving stock (Show, Eq)
 
-type TransferDescriptorParam =
-  ("fa2" :! Address, ("batch" :! [TransferDescriptor], "operator" :! Address))
+deriving anyclass instance IsoValue TransferDescriptorParam
+deriving anyclass instance HasAnnotation TransferDescriptorParam
+$(customGeneric "TransferDescriptorParam" $ withDepths
+    [ cstr @0
+      [ fld @1
+      , fld @2
+      , fld @2
+      ]
+    ]
+  )
 
 data FA2OwnerHook
```

Comment:
Can you please check whether hook-related types are used at all? I suspect they may be not used anymore because we are using default permission descriptor now which has "no hook" policy.

Reply:
`FA2OwnerHook` seems to only be used in a few tests in `Lorentz.Contracts.Test.FA2`

There's this comment next to those tests:

```
  ---- Owner hook test
  ----
  ---- Tests that validate that senders owner hook is not-called on transfer
  ---- using default origination options where both sender/receiver hooks
  ---- are set to be not called no matter what.
```

Reply:
Hm, okay, perhaps that test makes some sense. So I've created an issue that covers unused types as well as types present in `morley-ledgers`: https://github.com/tqtezos/stablecoin/issues/115